client/gutenberg/extensions/map/edit.js

```diff
@@ -89,26 +86,31 @@ class MapEdit extends Component {
 	};
 	apiCall( serviceApiKey = null, method = 'GET' ) {
 		const { noticeOperations } = this.props;
+		const { apiKey } = this.state;
 		const url = '/?rest_route=/jetpack/v4/service-api-keys/mapbox';
 		const fetch = serviceApiKey
 			? { url, method, data: { service_api_key: serviceApiKey } }
 			: { url, method };
-		apiFetch( fetch ).then(
-			result => {
-				noticeOperations.removeAllNotices();
-				this.setState( {
-					apiState: result.service_api_key ? API_STATE_SUCCESS : API_STATE_FAILURE,
-					apiKey: result.service_api_key,
-					apiKeyControl: result.service_api_key,
-				} );
-			},
-			result => {
-				this.onError( null, result.message );
-				this.setState( {
-					apiState: API_STATE_FAILURE,
-				} );
-			}
-		);
+		this.setState( { apiRequestOutstanding: true }, () => {
+			apiFetch( fetch ).then(
+				result => {
+					noticeOperations.removeAllNotices();
+					this.setState( {
+						apiState: result.service_api_key ? API_STATE_SUCCESS : API_STATE_FAILURE,
+						apiKey: result.service_api_key,
+						apiKeyControl: result.service_api_key,
+						apiRequestOutstanding: false,
+					} );
+				},
+				result => {
+					this.onError( null, result.message );
```

Comment:
is it normal that we always call `this.onError`?

Reply:
`this.onError` should only be called if the API call fails. Is that what you're asking about?